eng/common/templates/job/build-from-source.yml

```diff
@@ -0,0 +1,75 @@
+parameters:
+  jobNamePrefix: 'sourcebuild'    # Override for prefix for job names
+  managedOnly: false              # Specifies whether source is managed only. If so, one build platform is sufficient
+  portableOnly: false             # If true, builds one leg with portable flag enabled
+  prebuiltCheck: true             # Specifies whether to enforce prebuilt checking and fail if new prebuilts are introduced
+
+jobs:
+- template: /eng/common/templates/jobs/jobs.yml
```

Comment:
I'm not sure we can use this template like this... it has its own `publish-build-assets`, so if the repo itself uses `jobs.yml`, it might run this twice:

https://github.com/dotnet/source-build-reference-packages/blob/50dfd943f6e188f47fe138c1c43f2832e1b09555/eng/common/templates/jobs/jobs.yml#L46-L48

From a very small sample I see that some repos don't use jobs.yml (runtime, sdk), but others do (xliff-tasks).

I think we need to just use `job.yml`. Repos that use `jobs.yml` can *maybe* plug it in, or we can implement automatic setup in that case and pipe through the right variables. But the general case, I think, is someone using our template as a job(s) template in their build stage, where they've already set up this publish job.

Reply:
In SBRP this would look like the `azure-pipelines/builds/ci.yml` still using `jobs.yml`, but also including a source-build job.

And now that I think about it a bit more, I think we do need to make sure our job can be threaded into `jobs.yml` if a repo is using it, because `jobs.yml` sets up the job dependency for `publish-build-assets.yml` using the jobs that are passed into it. (If we don't get source-build in that list, the build may publish before we're done and skip our artifacts.)

Reply:
This isn't something that we need to do, though.  As long as repo owners include our template as a job when including `jobs.yml`, we should be OK, right?  Our template just needs to provide the jobs.

Reply:
> This isn't something that we need to do, though.

Sorry, what does "this" mean here?

> As long as repo owners include our template as a job when including `jobs.yml`, we should be OK, right?

As is, I think this PR would end up including `jobs.yml` twice. Once when the repo uses it, then again when this source-build template uses it. So they end up with something like this:

* ci.yml
  * jobs.yml
    * [ordinary jobs for Microsoft build]
    * build-from-source.yml
      * jobs.yml
        * centos7 build from source
        * **publish-build-assets.yml** 
    * **publish-build-assets.yml** 

The duplicate is a problem, we should just make a template that provides jobs:

* ci.yml
  * jobs.yml
    * [ordinary jobs for Microsoft build]
    * build-from-source.yml
      * centos7 build from source
    * publish-build-assets.yml 

(I don't actually know if you can use a template like that--to add jobs to the `jobs.yml` template's parameter list. So we might need to modify `jobs.yml` to bridge the gap.)

But we also need to support repos that don't use jobs.yml. They necessarily have `publish-build-assets.yml ` set up already because Arcade requires it, so we shouldn't add it again.

* ci.yml
  * [ordinary jobs for Microsoft build]
  * build-from-source.yml
    * centos7 build from source
  * publish-build-assets.yml

Reply:
Yes.  That's what I meant when I said, "Our template just needs to provide the jobs."  I just stated it terribly and confusingly! :(

Reply:
Hah, ok, perfect. 😄 Without context I thought your comment was meant as justification for the current impl, so I assumed some deep misunderstanding that wasn't there.